### ISSUE TITLE
Indian ISP's injecting Ads breaks site

### ISSUE THREAD

**Issue by ashojha21** [FreeCodeCamp/FreeCodeCamp#5999]
Challenge [Waypoint: Say Hello to HTML Elements](http://www.freecodecamp.com/challenges/waypoint-say-hello-to-html-elements#?solution=%3Ch1%3EHello%20%3C%2Fh1%3E%0A) has an issue.
User Agent is: <code>Mozilla/5.0 (Windows NT 10.0; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/47.0.2526.106 Safari/537.36</code>.
Please describe how to reproduce this issue, and include links to screenshots if possible.

My code:
```html
<h1>Hello World</h1>

```
After  I click run test. The output is not displayed on the right screen. I refreshed it multiple times, tried in different browser. Didn't help.
Please help me out of this. I am new to FCC.

**Comment by bugron** [comment]
@ashojha21 thanks for posting this issue!
I can't reproduce your issue neither in latest FF, nor in Chrome. After clicking "Run tests" button I can see the word "Hello" in iPhone frame. I can only repro the issue reported in https://github.com/FreeCodeCamp/FreeCodeCamp/issues/5973. Please, provide a screenshot/more description, thanks!

>***
>Please review the [**Guidelines for Contributing**](https://github.com/FreeCodeCamp/FreeCodeCamp/blob/staging/CONTRIBUTING.md), thank you!.
>***

**Comment by ashojha21** [comment]
![code2](https://cloud.githubusercontent.com/assets/13873005/12218964/711a7ffa-b756-11e5-955d-0c944b43153b.JPG)
![code3](https://cloud.githubusercontent.com/assets/13873005/12218966/74a7f1ac-b756-11e5-9f24-7d12381b62b9.JPG)

**Comment by ashojha21** [comment]
When I click on run test it doesnt display anything on the iphone. These above two pics are only thing i can give you.

**Comment by bugron** [comment]
@ashojha21 are you behind any proxy, anonymizing soft? Are you sure that popular CDN's are not blocked in your country because I can't reproduce this issue.

**Comment by ashojha21** [comment]
how to check if i am behind any proxy?

**Comment by bugron** [comment]
Search the same in Google.

**Comment by ashojha21** [comment]
no proxy server detected

**Comment by ashojha21** [comment]
anyone who can help?

**Comment by ltegman** [comment]
I can't say for sure, because I'm not able to repro the problem, but I think this may be solved by #5960 since it is solved the Vimeo CSP problem that is showing up in your console.

**Comment by BerkeleyTrue** [comment]
Looks like you are getting man-in-the-middle-ed.

http://203.94.243.63:3000/ <- this IP isn't us.
http://203.94.243.63:3000/nadip?json={url: "bower_components", catagory: ..., reputation: ..., } <- someone is attempting to log what sites you are using.

Sounds like a extension or something is tracking you, and doing a horrible job of it, too, because it is breaking the site.

Where are you located?

Here is the location of that IP:

![ip_location_finder_-_geolocation](https://cloud.githubusercontent.com/assets/6775919/12223919/fd7e897c-b798-11e5-866f-e143d0504000.png)

**Comment by ashojha21** [comment]
I am located in Delhi, INDIA. And my ISP is Mahanagar Telephone Nigam Ltd(MTNL)
I asked one of my friend who lives nearby and has internet connection from the same ISP MTNL, she is also having the same problem as mine. Does this mean my ISP is tracking my activity?
And what is the solution for this?

**Comment by SaintPeter** [comment]
1) Try connecting via an "incognito" tab.  I believe this disables all your extensions, so that should rule out a rogue browser plugin.
2) Run software like Malware Bytes or other anti-mailware software to see if you have some sort of malware installd.
3) If you are otherwise clean and suspect your ISP (since you friend has the same issue, it's probably the ISP) you may be able to connect through a VPN, which will encrypt your communications.

@BerkeleyTrue If it's his ISP, will connecting via SSH help, once we have it?

**Comment by BerkeleyTrue** [comment]
@SaintPeter Not entirely sure. It might make the site completely unusable if indeed they are being proxied at the ISP level as their browser would report that the SSL is invalid.

Another user (#5960) is also reporting this issue. They are located in Mumbai.

**Comment by BerkeleyTrue** [comment]
@ashojha21 Can you tell if that IP address is being called from any other site? You'd have to check the network tab in chrome devtools.

![new_tab](https://cloud.githubusercontent.com/assets/6775919/12225855/0cd3ffae-b7c8-11e5-9814-85b404618a67.png)

**Comment by ashojha21** [comment]
what exactly am i supposed to look for in the network tab?Should I go to any other website and see the network tab? How can i tell if IP is being called from other website?

**Comment by ashojha21** [comment]
![coursera](https://cloud.githubusercontent.com/assets/13873005/12226241/00ba845a-b83f-11e5-8ca3-22f778224a89.JPG)
This one is coursera.com

![freecodecamp](https://cloud.githubusercontent.com/assets/13873005/12226245/0ea3cbc6-b83f-11e5-936a-c4b016d5c93c.JPG)
freecodecamp.com

**Comment by AkiraLaine** [comment]
@ashojha21 Since you are using Chrome, I would suggest using a VPN. This should normally fix your problem. [Browsec](https://chrome.google.com/webstore/detail/browsec/omghfjlpggmjjaagoclmmobgdodcjboh?utm_source=chrome-ntp-icon) is VPN extension that I use and works. Try to download it and see if that solves your issue.

**Comment by abhisekp** [comment]
@ashojha21 I've a suggestion too and I don't know how far it will solve the issue but try changing your DNS to Google Public DNS **`8.8.8.8`** (primary) and **`8.8.4.4`** (secondary).

Hope this helps :smile:

**Comment by ryzokuken** [comment]
So, lastly. I think you guys could very well close this issue because it is not a lapse on your part, but just the govt. being a d***. This problem, however seems to go away as soon as you mark your location / identity. 

Both the Tor network / browser or the Browsec plugin would accomplish this easily. However, prefer Browsec, as: 
1. it can be disabled and enabled easily.
2. it allows you to use good'ol Chrome.
3. Tor seems dark to many people.
(happy? @AkiraLaine) although the solution could be preserved for reference in case someone gets this problem later. (:3)

**Comment by BerkeleyTrue** [comment]
> but just the govt. being a d***.

@ryzokuken Can you provide some details/sources on this?

**Comment by raisedadead** [comment]
@BerkeleyTrue : I can shed some light on this. Well for starters there is a messy censorship war going on, with Govt. of India (GOI) involved. And their warriors on the front line are GOI run ISP's like MTNL and others. Of course being a democracy, the decision has more been in favor of general public in recent times, after mass outrages.

I'll share a few links at the bottom of this post if you want to look into this, but setting the perspective:

GOI ordered via Court directive to block access to about 200 sites in Aug 2015, which included outages of some sites like **Github** too, a good petition ran on hacker news about this : [Help HN, lets make GitHub accessible again from India](https://news.ycombinator.com/item?id=8822332).

MTNL and other shady ISP's use some heavy proxy-ing and caching, which is actually good for content delivery, but turns out they unfortunately this to track subscribers and serve them up with pop-up ads and what not.

Plus one thing about GOI owned ISP is that people are sometimes really slog in doing even wrong things correctly. See the irony that they screw up in doing things that may not be justified. For example, you mentioned:
> Sounds like a extension or something is tracking you, and doing a horrible job of it, too, because it is breaking the site.

That's the best example I can give you. They can't even sometimes hide the injustice they do.

Anyways here are some links that might (NOT) interest you:
1. https://en.wikipedia.org/wiki/Internet_censorship_in_India#2015
2. http://www.medianama.com/2014/07/223-world-cup-2014-472-websites-including-google-docs-blocked-in-india-following-sony-complaint/
3. http://www.yogeshsarkar.com/blog/2014/04/12/how-to-block-mtnl-popups/
4. http://timesofindia.indiatimes.com/tech/tech-news/Over-200-sites-blocked-in-India-after-Sonys-piracy-complaint-Report/articleshow/37961214.cms?

Also this is not limited to GOI's ISPs, but some private telecom companies too. However, there are always work arounds, :wink: !

**Comment by BerkeleyTrue** [comment]
@raisedadead Thanks for the summery. I've update the title accordingly.

**Comment by SaintPeter** [comment]
@raisedadead Could you write up for or point our Indian campers to a more detailed guide on setting up or using `Browsec`?

**Comment by raisedadead** [comment]
@BerkeleyTrue Happy to help!

@SaintPeter I can write a wiki article if that might help, for using `Browsec`, well however it would be great if some one who actually uses one of the ISP's that are causing the trouble test the guide, so that its actually resolves the issues. Will do the needful asap.

**Comment by AkiraLaine** [comment]
@raisedadead @SaintPeter 
I could write up an article for Browsec. It's really simple. I suggested @ryzokuken to use Browsec as he had the same problem, and it worked for him.

**Comment by raisedadead** [comment]
@AkiraLaine That sound's great!

**Comment by ryzokuken** [comment]
@raisedead I could surely help as a lab rat.

@BerkeleyTrue The govt. has been banning websites for quite sometime now, also tracking our internet usage and what not. Recently, Facebook and other capitalists have started lounging for Free Basics, basically a filtergate that would classify internet and limit access. It would be kinda like, 5-8 sites for free, and basically a paid subscription kinda thing to access Github, FCC and stuff. Would be a really dark day if the government ceded to the pressure.

**Comment by re4ect** [comment]
@ryzokuken  in the u.s ISP's want to do something similar to that.

**Comment by raisedadead** [comment]
@AkiraLaine are you coming up with the wiki or should I do it?

**Comment by AkiraLaine** [comment]
@raisedadead Sorry, haven't had time to do it. Go ahead :smile:

**Comment by raisedadead** [comment]
Wiki article is available [here](https://github.com/FreeCodeCamp/FreeCodeCamp/wiki/Using-browsec-for-securing-your-connection-to-fcc)

**Comment by AkiraLaine** [comment]
@raisedadead *Hey what about other browsers, Internet Explorer, Safari, e.t.c.?* IE doesn't support extensions. Not sure about Safari, though. Edge will support them, but not anytime soon. 
I would suggest adding this to the article for people who don't have/don't want to install Chrome: [Hotspot Shield](https://www.hotspotshield.com/) - it's probably the only "software" VPN that has a free version that's unlimited. I've used most VPN's and this is the best "free" one out there

**Comment by raisedadead** [comment]
@AkiraLaine Yup, that would be cool. 

>  IE doesn't support extensions.

I agree, not extensions per se, but in IE's context they are called as ActiveX Plugins, I haven't checked for sure but there are VPN clients that use ActiveX technology to interact with IE, anyways that's out of scope of this issue or topic.

However, I think we can add the "Hotspot Sheild" to the alternate client list just like I mentioned "Tor".

Can you help me out on this one though, with your suggestion to the text along with links to the VPN client you already mentioned, I'll make the necessary changes as you propose. 

Thing is I haven't used this so, it better if you can supply the text, I'll make the changes.

Thanks a lot.

**Comment by BerkeleyTrue** [comment]
Closing as there is no further action we can take on our end to prevent this.

Happy Coding!

**Comment by mdelgadonyc** [comment]
Hi, I am mentioning my issue here because like @ashojha21 I am also seeing from Chrome a warning of a "Content Security Policy directive" violation although in my case I am attempting to access the Wiki page from within FCC. Also, I am accessing the site from NYC using Cablevision's Optimum Internet service.

I have no problem accessing Chat from the menu, but when I click on the Wiki menu link a blank page appears and in Chrome's Console log I get the error: vendor-main-066508b245.js:3 Refused to frame 'http://freecodecamp.github.io/wiki/en/' because it violates the following Content Security Policy directive: "frame-src *.gitter.im *.gitter.im https: *.vimeo.com *.twitter.com *.ghbtns.com *.freecatphotoapp.com freecodecamp.github.io 'self'".

![fcc-wiki-warning-screenshot from 2016-03-17 12 26 06](https://cloud.githubusercontent.com/assets/17136771/13853578/ae0efd8e-ec3d-11e5-850c-88dca127780c.png)

I first noticed the issue using Chrome 49.0.2623.87 on my desktop running Fedora 20. I tried to access it from the same desktop using Firefox and the page was also blank. I get a blank page as well from my Chromebook (on the same network).

I did install the Browsec Chrome plugin as was suggested to @ashojha21 but even with that activated the problem persists.

I can access the FCC Wiki http://freecodecamp.github.io/wiki/en/ directly with no problem. Any suggestions as to what might be causing this? Thanks

-Moisés
